### PR TITLE
3383bw3

### PR DESCRIPTION
**Pull request by katarinasulkic**

### CODE REVIEW

File: front/src/gc-components/GCListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { List, MantineNumberSize } from "@mantine/core";
```
Comment (Logovas1994) [inline front/src/gc-components/GCListItem.tsx]: If you are not using MantineNumberSize then delete it

File: front/src/pages/nft/[id].tsx [hunk 2]
```diff
@@ -22,13 +22,25 @@ const SingleNFTPage = (props: SingleNFTPageProps) => {
...
+            <>
+                <GCGrid className={classes.grid}>
+                    <GCGridCol sm={12} md={6} className={classes.position}>
+                        <div className={classes.nftImage}>
+                            <GCImage src={data.nftImageUrl} alt="gun" />
+                        </div>
+                        <div className={classes.positionText}>
+                            <GCText>{data.name}</GCText>
+                            <div className={classes.listSize}>
+                                <GCList size={12}>
```
Comment (Logovas1994) [inline front/src/pages/nft/[id].tsx]: Please take another look at the layout from Miro. We have a list component. It should contain the following listitem: mintingDate, rarity, collection, firstOwner. All of them are in the data from the request.

Separately, on our visualization there is a price. It wouldn't be bad to take it out separately.

What about the description of the gun , it should be a separate text component under list component.

File: front/src/gc-components/GCListItem.tsx (hunk 1 above)
Comment (katarinasulkic) [reply front/src/gc-components/GCListItem.tsx]: I deleted

File: front/src/pages/nft/[id].tsx (hunk 2 above)
Comment (katarinasulkic) [reply front/src/pages/nft/[id].tsx]: I finished that

File: front/src/gc-components/GCList.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { DefaultMantineColor, List, MantineNumberSize } from "@mantine/core";
+import { ReactNode } from "react";
+
+interface GCListProps {
+    children: ReactNode;
+    size?: MantineNumberSize;
+    color?: DefaultMantineColor;
+}
```
Comment (AbrosimovVladislav) [inline front/src/gc-components/GCList.tsx:13]: Each GC wrapper component props should be extended from original mantine component props, in this case from ListProps
Plz check how we doind this on example of GCLink

File: front/src/gc-components/GCList.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+export const GCList = ({ size, color, children }: GCListProps) => {
```
Comment (AbrosimovVladislav) [inline front/src/gc-components/GCList.tsx]: to the props list we also should add ...props, this gives to us possibilty to support all props from Mantine original component
Plz check how we doind this on example of GCLink

File: front/src/gc-components/GCList.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    return (
+        <List size={size} color={color}>
```
Comment (AbrosimovVladislav) [inline front/src/gc-components/GCList.tsx]: Here we also should use props in usuing original Mantine component
Plz check how we doind this on example of GCLink

File: front/src/gc-components/GCListItem.tsx [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { List } from "@mantine/core";
+import { ReactNode } from "react";
+
+interface GCListItemProps {
```
Comment (AbrosimovVladislav) [inline front/src/gc-components/GCListItem.tsx]: The same stuff as in GCList component and props

File: front/src/gc-components/GCText.tsx [hunk 7]
```diff
@@ -1,20 +1,20 @@
-import {ReactNode} from "react";
-import {DefaultMantineColor, MantineNumberSize, Sx, Text, TextProps} from "@mantine/core";
+import { ReactNode } from "react";
```
Comment (AbrosimovVladislav) [inline front/src/gc-components/GCText.tsx:1]: Are just code styles changed here? You should use the same prettier settings as everybody in team. @Logovas1994 plz help here

File: front/src/pages/nft/[id].tsx (hunk 2 above)
Comment (AbrosimovVladislav) [reply front/src/pages/nft/[id].tsx]: I agree with Peter, and lets create separate component like nft info for this. @Logovas1994 plz help here

Comment (AbrosimovVladislav) [reply front/src/pages/nft/[id].tsx]: Also we should add name of field that is inside info like, 
Minting date 01.01.2023
Rarity USUAL
etc...

File: front/src/pages/nft/SingleNFTPageStyles.ts [hunk 8]
```diff
@@ -2,6 +2,36 @@ import { createStyles } from "@mantine/core";
 
 export const useStyles = createStyles((theme) => ({
     grid: {
-        padding: [`0 ${theme.spacing.xl}px 0 ${theme.spacing.xl}px`]
+        padding: [`0 ${theme.spacing.xl}px 0 ${theme.spacing.xl}px`],
```
Comment (nastyakuzusha) [inline front/src/pages/nft/SingleNFTPageStyles.ts]: what is sense of using this shorthand? you could easily replace it with 0 ${theme.spacing.xl}px - it just describes that top and bottom are 0 and left-right are xl

File: front/src/components/nft-card-information/NftCardInformationStyle.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import { createStyles } from "@mantine/core";
+
+export const useStyles = createStyles(() => ({
+    textBold: {
+        fontWeight: 400,
+    },
+
+    listSize: {
+        margin: "20px 0 20px;",
```
Comment (nastyakuzusha) [inline front/src/components/nft-card-information/NftCardInformationStyle.ts]: pls use theme variables (with a shorthand property)

File: front/src/pages/nft/[id].tsx [hunk 10]
```diff
@@ -22,13 +23,20 @@ const SingleNFTPage = (props: SingleNFTPageProps) => {
...
     if (isSuccess) {
-        return <>
-            <GCGrid className={classes.grid}>
-                <GCGridCol sm={12} md={6}>
-                    <GCImage src={data.nftImageUrl} alt="gun" />
-                </GCGridCol>
-            </GCGrid>
-        </>;
+        return (
+            <>
```
Comment (nastyakuzusha) [inline front/src/pages/nft/[id].tsx]: you dont need an empty tag here

File: front/src/pages/nft/[id].tsx [hunk 11]
```diff
@@ -22,13 +23,20 @@ const SingleNFTPage = (props: SingleNFTPageProps) => {
...
+                <GCGrid className={classes.grid}>
+                    <GCGridCol sm={6} md={6}>
+                        <div>
```
Comment (nastyakuzusha) [inline front/src/pages/nft/[id].tsx]: and you dont need div here

File: front/.prettierrc.json [hunk 12]
```diff
@@ -1,8 +1,7 @@
...
-  "tabWidth": 4,
-  "semi": true,
-  "singleQuote": false,
-  "printWidth": 120
-
+    "trailingComma": "es5",
+    "tabWidth": 4,
+    "semi": true,
+    "singleQuote": false,
+    "printWidth": 120
```
Comment (nastyakuzusha) [inline front/.prettierrc.json]: remove this file from git please

File: front/src/components/nft-card-information/NftCardInformation.tsx [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+interface NftCardInformationProps {
+    data: NFTCard;
+}
+
+const NftCardInformation = ({ data }: NftCardInformationProps) => {
+    const { classes } = useStyles();
+
+    return (
+        <div>
+            <GCText className={classes.textBold}>{data.name}</GCText>
```
Comment (nastyakuzusha) [inline front/src/components/nft-card-information/NftCardInformation.tsx]: dont we have this bold property in GCText props, do we?

File: front/src/components/nft-card-information/NftCardInformationStyle.ts (hunk 9 above)
Comment (katarinasulkic) [reply front/src/components/nft-card-information/NftCardInformationStyle.ts]: Is it correct write like this?

margin: [`20px 0 ${theme.spacing.xl}px`],

File: front/src/components/nft-card-information/NftCardInformation.tsx [hunk 14]
```diff
@@ -0,0 +1,33 @@
+import { GCList, GCListItem, GCText } from "../../gc-components";
```
Comment (nastyakuzusha) [inline front/src/components/nft-card-information/NftCardInformation.tsx:1]: pls make sure layout is somehow related to design
you can find template here https://www.figma.com/file/k7cewrEXHk3Ik7Pqftgvff/Untitled?node-id=907%3A3777&t=8hNY2AEqhGCIkAv4-1

Comment (katarinasulkic) [reply front/src/components/nft-card-information/NftCardInformation.tsx:1]: I didn't know there was a design. Do I need to make the same as here?

**Comment by AbrosimovVladislav** [comment]
@katarinasulkic why prettier.json file is deleted?

File: front/src/components/nft-card-information/NftCardInformation.tsx (hunk 14 above)
Comment (AbrosimovVladislav) [reply front/src/components/nft-card-information/NftCardInformation.tsx:1]: Nope, I will create separate task for draft design

File: front/src/gc-components/GCGridCol.tsx [hunk 15]
```diff
@@ -15,7 +15,7 @@ interface GCGridColProps extends GridProps{
 
 export const GCGridCol = ({ className, children, span, xs, sm, md, lg, xl, ...props }: GCGridColProps) => {
     return (
-        <Grid.Col className= {className} span={span} xs={xs} sm={sm} md={md} lg={lg} xl={xl} {...props}>
+        <Grid.Col className={className} span={span} xs={xs} sm={sm} md={md} lg={lg} xl={xl} {...props}>
```
Comment (AbrosimovVladislav) [inline front/src/gc-components/GCGridCol.tsx:18]: I think all that changed here is just code style @Logovas1994 plz control that prettier works well, and this is write code style

File: front/src/gc-components/GCList.tsx (hunk 3 above)
Comment (AbrosimovVladislav) [reply front/src/gc-components/GCList.tsx:13]: @katarinasulkic it is still not extending from ListProps (it is props for original mantine component)
final solution should looks like this
Снимок экрана 2022-11-18 в 20 29 01

File: front/src/gc-components/GCListItem.tsx (hunk 6 above)
Comment (AbrosimovVladislav) [reply front/src/gc-components/GCListItem.tsx]: Same like in List, it is still not extending anything

**Review by AbrosimovVladislav — CHANGES_REQUESTED** [review]
Some of changes still actual, plz refactor it one more time

File: front/src/components/nft-card-information/NftCardInformationStyle.ts (hunk 9 above)
Comment (AbrosimovVladislav) [reply front/src/components/nft-card-information/NftCardInformationStyle.ts]: @nastyakuzusha plz answer

File: front/.prettierrc.json (hunk 12 above)
Comment (AbrosimovVladislav) [reply front/.prettierrc.json]: wtf, why we need to remove it? 
we added it to have same code style for whole team, why why we delete it :-D

Comment (AbrosimovVladislav) [reply front/.prettierrc.json]: Plz return it :)

**Pull request merged**